controllers/lease_controller.go

```diff
@@ -0,0 +1,544 @@
+		env = append(env, getEnvVarFromSecrets("OS_USERNAME", storeValues.SecretRef.Name, "username"))
+		env = append(env, getEnvVarFromSecrets("OS_PASSWORD", storeValues.SecretRef.Name, "password"))
+		env = append(env, getEnvVarFromSecrets("OS_TENANT_NAME", storeValues.SecretRef.Name, "tenantName"))
+	}
+
+	if provider == "OSS" {
+		env = append(env, getEnvVarFromSecrets("ALICLOUD_ENDPOINT", storeValues.SecretRef.Name, "storageEndpoint"))
+		env = append(env, getEnvVarFromSecrets("ALICLOUD_ACCESS_KEY_SECRET", storeValues.SecretRef.Name, "accessKeySecret"))
+		env = append(env, getEnvVarFromSecrets("ALICLOUD_ACCESS_KEY_ID", storeValues.SecretRef.Name, "accessKeyID"))
+	}
+
+	if provider == "ECS" {
+		env = append(env, getEnvVarFromSecrets("ECS_ENDPOINT", storeValues.SecretRef.Name, "endpoint"))
+		env = append(env, getEnvVarFromSecrets("ECS_ACCESS_KEY_ID", storeValues.SecretRef.Name, "accessKeyID"))
+		env = append(env, getEnvVarFromSecrets("ECS_SECRET_ACCESS_KEY", storeValues.SecretRef.Name, "secretAccessKey"))
+	}
+
+	if provider == "OCS" {
+		env = append(env, getEnvVarFromSecrets("OCS_ENDPOINT", storeValues.SecretRef.Name, "endpoint"))
+		env = append(env, getEnvVarFromSecrets("OCS_ACCESS_KEY_ID", storeValues.SecretRef.Name, "accessKeyID"))
+		env = append(env, getEnvVarFromSecrets("OCS_SECRET_ACCESS_KEY", storeValues.SecretRef.Name, "secretAccessKey"))
+		env = append(env, getEnvVarFromSecrets("OCS_REGION", storeValues.SecretRef.Name, "region"))
+		env = append(env, getEnvVarFromSecrets("OCS_DISABLE_SSL", storeValues.SecretRef.Name, "disableSSL"))
+		env = append(env, getEnvVarFromSecrets("OCS_INSECURE_SKIP_VERIFY", storeValues.SecretRef.Name, "insecureSkipVerify"))
+	}
+
+	return env
+}
+
+func getEnvVarFromValues(name, value string) v1.EnvVar {
+	return v1.EnvVar{
+		Name:  name,
+		Value: value,
+	}
+}
+
+func getEnvVarFromFields(name, fieldPath string) v1.EnvVar {
+	return v1.EnvVar{
+		Name: name,
+		ValueFrom: &v1.EnvVarSource{
+			FieldRef: &v1.ObjectFieldSelector{
+				FieldPath: fieldPath,
+			},
+		},
+	}
+}
+
+func getEnvVarFromSecrets(name, secretName, secretKey string) v1.EnvVar {
+	return v1.EnvVar{
+		Name: name,
+		ValueFrom: &v1.EnvVarSource{
+			SecretKeyRef: &v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key: secretKey,
+			},
+		},
+	}
+}
+
+func getCompactJobCommands(etcd *druidv1alpha1.Etcd) []string {
+	command := []string{"" + "etcdbrctl"}
+	command = append(command, "compact")
+	command = append(command, "--data-dir=/var/etcd/data")
+	command = append(command, "--snapstore-temp-directory=/var/etcd/data/tmp")
+	command = append(command, "--enable-snapshot-lease-renewal=true")
+	command = append(command, "--full-snapshot-lease-name="+getFullLease(etcd))
+	command = append(command, "--delta-snapshot-lease-name="+getDeltaLease(etcd))
+
+	var quota int64 = 8 * 1024 * 1024 * 1024 // 8Gi
```

Comment:
Can we fetch this value from a constant instead of configuring it here within the reconciliation logic?

Reply:
Done